package/src/components/Message/MessageSimple/MessageSimple.tsx

```diff
@@ -175,6 +201,102 @@ const MessageSimpleWithContext = <
 
   const repliesCurveColor = isMessageReceivedOrErrorType ? grey_gainsboro : backgroundColor;
 
+  const translateX = useSharedValue(0);
+  const touchStart = useSharedValue<{ x: number; y: number } | null>(null);
+
+  const onSwipeToReply = () => {
+    clearQuotedMessageState();
+    setQuotedMessageState(message);
+  };
+
+  const THRESHOLD = 25;
+
+  const swipeGesture = Gesture.Pan()
+    .manualActivation(true)
+    .onBegin((event) => {
+      touchStart.value = { x: event.x, y: event.y };
+    })
+    .onTouchesMove((event, state) => {
+      if (!touchStart.value || !event.changedTouches.length) {
+        state.fail();
+        return;
+      }
+
+      const xDiff = Math.abs(event.changedTouches[0].x - touchStart.value.x);
+      const yDiff = Math.abs(event.changedTouches[0].y - touchStart.value.y);
+      const isHorizontalPanning = xDiff > yDiff;
+
+      if (isHorizontalPanning) {
+        state.activate();
+      } else {
+        state.fail();
+      }
+    })
+    .onStart(() => {
+      translateX.value = 0;
+    })
+    .onChange(({ translationX }) => {
+      if (translationX > 0) {
+        translateX.value = translationX;
+      }
+    })
+    .onEnd(() => {
+      if (translateX.value >= THRESHOLD) {
+        runOnJS(onSwipeToReply)();
+        runOnJS(triggerHaptic)('impactMedium');
+      }
+      translateX.value = withSpring(0, {
+        dampingRatio: 1,
+        duration: 500,
+        overshootClamping: true,
+        stiffness: 1,
+      });
+    });
+
+  const messageBubbleAnimatedStyle = useAnimatedStyle(() => ({
+    transform: [{ translateX: translateX.value }],
+  }));
+
+  const swipeContentAnimatedStyle = useAnimatedStyle(() => ({
+    opacity: interpolate(translateX.value, [0, THRESHOLD], [0, 1]),
+    transform: [
+      {
+        translateX: interpolate(
+          translateX.value,
+          [0, THRESHOLD],
+          [-THRESHOLD, 0],
+          Extrapolation.CLAMP,
+        ),
+      },
+    ],
+  }));
+
+  const renderMessageBubble = (
```

Comment:
This is a potential performance issue. Both of these components will be recreated on every `MessageSimple` render; which as we know can be relatively heavy

Reply:
I have added useMemo with suitable dependencies, and that should be fine. Let me know what do you think.